src/server/game/Entities/Object/Object.cpp

```diff
@@ -3028,11 +3028,13 @@ bool WorldObject::SetInPhase(uint32 id, bool update, bool apply)
         {
             // if area phase passes the condition we should not remove it (ie: if remove called from aura remove)
             // this however breaks the .mod phase command, you wont be able to remove any area based phases with it
-            if (std::vector<PhaseInfoStruct> const* phases = sObjectMgr->GetPhasesForArea(GetAreaId()))
-                for (PhaseInfoStruct const& phase : *phases)
-                    if (id == phase.Id)
-                        if (sConditionMgr->IsObjectMeetToConditions(this, phase.Conditions))
-                            return false;
+            PhaseInfo const& phases = sObjectMgr->GetAreaAndZonePhases();
+            for (PhaseInfo::const_iterator itr = phases.begin(); itr != phases.end(); ++itr)
```

Comment:
No, I mean something like `map<phaseId, vector<PhaseInfoStruct const*>>` in ObjectMgr (constructed in same place as `_phases`)